### ISSUE TITLE
"error should be the last type when returning multiple items"

### ISSUE THREAD

**Issue by metaleap** [golang/lint#286]
Except it *is*, "technically":

    func cmdTryStart (name string, args ...string) (runerr error, killerr error) {
        ....

Amusing edge-case, but goes to show you might wanna double-check that the function does in fact contradict the warning about to be uttered  ;)

**Comment by leonelquinteros** [comment]
I'm willing to implement this fix. 

Just a question, how should be the allowance for multiple error returns? Should all error values be returned always at the end? 

So: 

```go
func OK() (a int, b int, c error, d error) {
    //...
}
```

Would be OK. 

And: 

```go
func FAIL() (a int, b error, c int, d error) {
    //...
}
```

Should fail? 

Or as long as the last one is an error we're OK?
